torch/lib/TensorLib/HalfConvert.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+namespace tlib {
+
+  template<typename To, typename From>
+  static inline To HalfFix(From h) {
+    return To { h.x };
+  }
```

Comment:
There are three half types THHalf, tlib::Half and cuda's __half/half that are all just a struct with a short x. Cuda's half need cuda present and tlib doesn't expose TH in its header file, so we need to convert between these half types.